cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -998,6 +1002,88 @@ impl Node<Option<cst::Add>> {
     fn to_expr(&self) -> Result<ast::Expr> {
         self.to_expr_or_special()?.into_expr()
     }
+
+    // Peel the grammar onion until we see valid RHS
+    pub(crate) fn to_has_rhs(&self) -> Result<Either<SmolStr, NonEmpty<UnreservedId>>> {
+        let inner @ cst::Add { initial, extended } = self.try_as_inner()?;
+        let err = |loc| {
```

Comment:
no strong opinion on whether you do this or not, but if you annotate the return type here
```suggestion
        let err = |loc| -> ParseErrors {
```
then I think you won't need the type annotation on the `Ok` on line 1024

Reply:
Unfortunately it didn't work. We need to update the signature of another closure.